### DISCUSSION TITLE
File-Extensions on Mac/Safari

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#11543] [Q&A]
I'm developing a WASM-app that saves and loads data to the filesystem. That works everywhere as far as I can tell. But on Mac using Safari, I cannot load the files I've created if the have an "abnormal" file extension. Let's say I have "file.mytype", then I can see that file in the loading window, but I am unable to select and load it.

What could I do to resolve this? Do I need to register the filetype anywhere or allow it to be selected by Safari? Any ideas?

**Comment by jeromelaban** [comment]
Can you give more details on how you're loading the data from the filesystem?

**Reply by TopperDEL** [reply]
Basically it's this:

```
       private async void Load_Click(object sender, RoutedEventArgs e)
        {
            var fileOpenPicker = new FileOpenPicker();
            //fileOpenPicker.SuggestedStartLocation = PickerLocationId.ComputerFolder;
            fileOpenPicker.FileTypeFilter.Add(".myext");

            // For Uno.WinUI-based apps
            var hwnd = WinRT.Interop.WindowNative.GetWindowHandle(MainWindow);
            WinRT.Interop.InitializeWithWindow.Initialize(fileOpenPicker, hwnd);

            StorageFile pickedFile = await fileOpenPicker.PickSingleFileAsync();
            if (pickedFile != null)
            {
                // File was picked, you can now use it
                ViewModel.LoadTree(await FileIO.ReadTextAsync(pickedFile), true);
            }
            else
            {
                // No file was picked or the dialog was cancelled.
            }
        }
```

I can see my file "myfile.myext" on MacOS, but it is grayed out and I can't select it.

**Reply by jeromelaban** [reply]
This could very well be a Safari limitation. @MartinZikmund may know more.

**Reply by MartinZikmund** [reply]
Created an issue, need to investigate
